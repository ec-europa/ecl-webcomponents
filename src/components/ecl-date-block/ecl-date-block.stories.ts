const getArgs = () => {
  return {
    variant: 'ongoing',
    day: '26',
    month: 'Sep',
    monthFull: 'September',
    year: '2022',
    dateTime: '2019-09-26',
  };
};

const getArgTypes = () => {
  return {
    color_mode: { table: { disable: true } },
    variant: {
      type: { name: 'select' },
      options: ['ongoing', 'past'],
      description: "Date block variant",
      control: {
        labels: {
          ongoing: 'upcoming & ongoing',
          past: 'past & cancelled',
        },
      },
    },
    day: {
      type: { name: 'string'},
      description: 'The date day',
    },
    month: {
      type: { name: 'string'},
      description: 'The date month (abridged)',
    },
    monthFull: {
      name: 'full month',
      type: { name: 'string'},
      description: 'The date full month',
    },
    year: {
      type: { name: 'string'},
      description: 'The date year',
    },
    dateTime: {
      name: 'date-time',
      type: { name: 'string'},
      description: 'Day time (YYY-MM-DD)',
    },
  };
};

export default {
  title: 'Components/date-block',
};

const Template = (args) => 
  `<ecl-date-block
    variant="${args.variant}"
    day="${args.day}"
    month="${args.month}"
    month-full="${args.monthFull}"
    year="${args.year}"
    date-time="${args.dateTime}" 
  >
  </ecl-date-block>`;

export const DateBlock = Template.bind({});
DateBlock.storyName = 'default';
DateBlock.args = getArgs();
DateBlock.argTypes = getArgTypes();
