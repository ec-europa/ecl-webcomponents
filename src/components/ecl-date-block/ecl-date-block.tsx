import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-date-block',
  styleUrls: {
    ec: './build/styles/ecl-date-block-ec.css',
    eu: './build/styles/ecl-date-block-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclDateBlock {
  @Prop() styleClass: string = '';
  @Prop() variant: string = 'default';
  @Prop() theme: string = 'ec';
  @Prop() day: string;
  @Prop() month: string;
  @Prop() year: string;
  @Prop() monthFull: string;
  @Prop() dateTime: string;

  getClass(): string {
    return [
      `ecl-date-block`,
      `ecl-date-block--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <time class={this.getClass()}>
      { this.dateTime ? 
        <span class="ecl-date-block__daytime">
          {this.dateTime}
        </span> : '' }
      { this.day ?  
        <span
          class="ecl-date-block__day"
          aria-hidden="true"
          >
            {this.day}
        </span> : '' }
      { this.month ?
        <abbr
          title={this.monthFull}
          class="ecl-date-block__month"
          aria-hidden="true"
        >
          {this.month}
        </abbr> : '' }
      { this.year ?
        <span
          class="ecl-date-block__year"
          aria-hidden="true"
        >
          {this.year}
        </span> : '' }
      </time>
    )
  }
}
