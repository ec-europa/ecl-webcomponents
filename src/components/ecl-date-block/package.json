{
  "name": "@ecl/web-components-date-block",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.8.0",
  "description": "ECL Date Block",
  "main": "ecl-date-block.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.8.4",
    "@ecl/theme-ec": "3.8.4",
    "@ecl/theme-eu": "3.8.4",
    "@ecl/vanilla-component-date-block": "3.8.4",
    "@ecl/vanilla-utility-screen-reader": "3.8.4",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
