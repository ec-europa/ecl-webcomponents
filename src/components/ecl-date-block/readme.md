# ecl-date-block



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `dateTime`   | `date-time`   |             | `string` | `undefined` |
| `day`        | `day`         |             | `string` | `undefined` |
| `month`      | `month`       |             | `string` | `undefined` |
| `monthFull`  | `month-full`  |             | `string` | `undefined` |
| `styleClass` | `style-class` |             | `string` | `''`        |
| `theme`      | `theme`       |             | `string` | `'ec'`      |
| `variant`    | `variant`     |             | `string` | `'default'` |
| `year`       | `year`        |             | `string` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
