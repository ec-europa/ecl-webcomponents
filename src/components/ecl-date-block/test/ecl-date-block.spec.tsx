import { newSpecPage } from '@stencil/core/testing';
import { EclDateBlock } from '../ecl-date-block';

describe('ecl-date-block', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclDateBlock],
      html: `<ecl-date-block
              variant="default"
              theme="ec"
              day="24"
              month="Sep"
              year="2022"
              date-time="2022-09-24" 
            >
            </ecl-date-block>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
