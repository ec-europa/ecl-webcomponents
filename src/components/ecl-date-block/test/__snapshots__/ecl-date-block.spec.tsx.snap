// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-date-block renders 1`] = `
<ecl-date-block date-time="2022-09-24" day="24" month="Sep" month-full="September" variant="ongoing" year="2022">
  <time class="ecl-date-block ecl-date-block--ongoing" datetime="2022-09-24">
    <span class="ecl-date-block__daytime">
      2022-09-24
    </span>
    <span aria-hidden="true" class="ecl-date-block__day">
      24
    </span>
    <abbr aria-hidden="true" class="ecl-date-block__month" title="September">
      Sep
    </abbr>
    <span aria-hidden="true" class="ecl-date-block__year">
      2022
    </span>
  </time>
</ecl-date-block>
`;
